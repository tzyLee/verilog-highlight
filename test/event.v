module test;

parameter d = 50; // d declared as a parameter and
reg [7:0] r; // r declared as an 8-bit reg

initial begin // a waveform controlled by sequential delay
    #d r = 'h35;
    #d r = 'hE2;
    #d r = 'h00;
    #d r = 'hF7;
    #d -> end_wave; //trigger an event called end_wave
end


initial fork
    #50 r = 'h35;
    #100 r = 'hE2;
    #150 r = 'h00;
    #200 r = 'hF7;
    #250 -> end_wave[2];
join

endmodule