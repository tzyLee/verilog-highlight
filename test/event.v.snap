>module test;
#^^^^^^ source.verilog meta.module.verilog storage.type.module.verilog
#      ^ source.verilog meta.module.verilog
#       ^^^^ source.verilog meta.module.verilog entity.name.type.module.verilog
#           ^ source.verilog meta.module.verilog punctuation.section.module.begin.verilog
>
>parameter d = 50; // d declared as a parameter and
#^^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog keyword.other.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog variable.other.readwrite.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog
#            ^ source.verilog meta.module.verilog meta.module.body.verilog keyword.operator.assignment.verilog
#             ^ source.verilog meta.module.verilog meta.module.body.verilog
#              ^^ source.verilog meta.module.verilog meta.module.body.verilog constant.numeric.dec.verilog
#                ^ source.verilog meta.module.verilog meta.module.body.verilog punctution.terminator.statement.verilog
#                 ^ source.verilog meta.module.verilog meta.module.body.verilog
#                  ^^ source.verilog meta.module.verilog meta.module.body.verilog comment.line.double-slash.verilog punctuation.definition.comment.verilog
#                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog comment.line.double-slash.verilog
>reg [7:0] r; // r declared as an 8-bit reg
#^^^ source.verilog meta.module.verilog meta.module.body.verilog support.type.verilog
#   ^ source.verilog meta.module.verilog meta.module.body.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.item-aceess.verilog meta.item-index.verilog punctuation.definition.begin.bracket.verilog
#     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.item-aceess.verilog meta.item-index.verilog constant.numeric.dec.verilog
#      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.item-aceess.verilog meta.item-index.verilog punctuation.seperator.range.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.item-aceess.verilog meta.item-index.verilog constant.numeric.dec.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.item-aceess.verilog punctuation.definition.end.bracket.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog variable.other.readwrite.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog punctuation.terminator.statement.verilog
#            ^ source.verilog meta.module.verilog meta.module.body.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog comment.line.double-slash.verilog punctuation.definition.comment.verilog
#               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog comment.line.double-slash.verilog
>
>initial begin // a waveform controlled by sequential delay
#^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog keyword.control.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog
#        ^^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.other.block.verilog
#             ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#              ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog comment.line.double-slash.verilog punctuation.definition.comment.verilog
#                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog comment.line.double-slash.verilog
>    #d r = 'h35;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.delay.verilog
#     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.assign.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#           ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog
#               ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog punctuation.terminator.statement.verilog
>    #d r = 'hE2;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.delay.verilog
#     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.assign.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#           ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog
#               ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog punctuation.terminator.statement.verilog
>    #d r = 'h00;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.delay.verilog
#     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.assign.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#           ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog
#               ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog punctuation.terminator.statement.verilog
>    #d r = 'hF7;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.delay.verilog
#     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.assign.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#           ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog constant.numeric.hex.verilog
#               ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog punctuation.terminator.statement.verilog
>    #d -> end_wave; //trigger an event called end_wave
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.delay.verilog
#     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#       ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.operator.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#          ^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog variable.other.readwrite.verilog
#                  ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog punctuation.terminator.statement.verilog
#                   ^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog
#                    ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog comment.line.double-slash.verilog punctuation.definition.comment.verilog
#                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog comment.line.double-slash.verilog
>end
#^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.sequential-block.verilog keyword.other.block.verilog
>
>
>initial fork
#^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog keyword.control.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog
#        ^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.other.block.verilog
>    #50 r = 'h35;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.delay.verilog
#     ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.unsigned.verilog
#       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog variable.other.readwrite.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.assign.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#            ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#              ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog
#                ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog punctuation.terminator.statement.verilog
>    #100 r = 'hE2;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.delay.verilog
#     ^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.unsigned.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog variable.other.readwrite.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.assign.verilog
#            ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#               ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog
#                 ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog punctuation.terminator.statement.verilog
>    #150 r = 'h00;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.delay.verilog
#     ^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.unsigned.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog variable.other.readwrite.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.assign.verilog
#            ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#               ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog
#                 ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog punctuation.terminator.statement.verilog
>    #200 r = 'hF7;
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.delay.verilog
#     ^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.unsigned.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#         ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog variable.other.readwrite.verilog
#          ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.assign.verilog
#            ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#             ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog storage.type.number.verilog
#               ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.hex.verilog
#                 ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog punctuation.terminator.statement.verilog
>    #250 -> end_wave[2];
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.delay.verilog
#     ^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog constant.numeric.unsigned.verilog
#        ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#         ^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.operator.verilog
#           ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog
#            ^^^^^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog variable.other.readwrite.verilog
#                    ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog meta.item-aceess.verilog meta.item-index.verilog punctuation.definition.begin.bracket.verilog
#                     ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog meta.item-aceess.verilog meta.item-index.verilog constant.numeric.dec.verilog
#                      ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog meta.item-aceess.verilog punctuation.definition.end.bracket.verilog
#                       ^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog punctuation.terminator.statement.verilog
>join
#^^^^ source.verilog meta.module.verilog meta.module.body.verilog meta.parallel-block.verilog keyword.other.block.verilog
>
>endmodule
#^^^^^^^^^ source.verilog meta.module.verilog storage.type.module.verilog